src/sdk/trace/id_generator.rs

```diff
@@ -0,0 +1,25 @@
+//! Id Generator
+use crate::api;
+use rand::{rngs, Rng};
+use std::cell::RefCell;
+
+/// Generates Trace and Span ids
+#[derive(Clone, Debug, Default)]
+pub struct IdGenerator {}
```

Comment:
Users can currently construct this with `IdGenerator {}`. However, that means that if fields are added to this type in the future, any code that constructs it will break.

For forward-compatibility reasons, you may want to give this a private field, like so:
```suggestion
pub struct IdGenerator {
    _private: (),
}
```
This would mean users would have to use `IdGenerator::default()` or `IdGenerator::new()` (if one was added). If additional fields are added in the future, code that uses `default` will still work.